pkg/http/server/server.go

```diff
@@ -13,6 +14,17 @@ func New(cfg *Config) *http.Server {
 	}
 }
 
+func NewTLS(cfg *Config) *http.Server {
+	server := &http.Server{
+		Addr:        fmt.Sprintf("%s:%d", cfg.TLSHost, cfg.TLSPort),
+		IdleTimeout: cfg.IdleConnTimeout,
+		ReadTimeout: cfg.Timeout,
+	}
+	// This will NOT require peer cert during TLS handshake
+	server.TLSConfig = &tls.Config{ClientAuth: tls.VerifyClientCertIfGiven}
```

Comment:
I would be expecting config.TLSCert and config.TLSKey as could see from this example code https://gist.github.com/denji/12b3a568f092ab951456

Reply:
I have missed the TLS server part while picking, I will fix that

Reply:
We shouldn't since https server and auth might use different Authorities to validate certs and we mentioned we will `ClientAuth` customizable in the future